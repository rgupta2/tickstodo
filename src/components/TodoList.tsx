import * as React from "react";
import { ITodo } from "../entities/ITodo";
import * as styles from "./css/styles.css";
import { TodoItem } from "./TodoItem";

interface IProps {
    todos: ITodo[];
    onTodoClick: Function;
    onTrashClick: Function;
}

export class TodoList extends React.Component<any, any> {
    public shouldComponentUpdate(nextProps: IProps) {
        return this.props.todos !== nextProps.todos;
    }

    public render() {
        return (
            <div className={styles.reminders}>
                {this.props.todos.map((todo: ITodo) => {
                    return (
                        <TodoItem
                            key={todo.id}
                            {...todo}
                            onItemClick={() => this.props.onTodoClick(todo.id)}
                            onTrashClick={() => this.props.onTrashClick(todo.id)}
                        />
                    );
                })}
            </div>
        );
    }
}
