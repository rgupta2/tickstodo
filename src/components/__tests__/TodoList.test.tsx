
import * as React from "react";
import * as renderer from "react-test-renderer";
import ITodo from "../../entities/ITodo";
import { TodoList } from "../TodoList";

it("renders correctly", () => {
    let todo: ITodo = {
        completed: false,
        id: 1,
        text: "some todo",
    };
    const tree = renderer.create(
        <TodoList todos={[todo]} onTodoClick={jest.fn} onTrashClick={jest.fn}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
