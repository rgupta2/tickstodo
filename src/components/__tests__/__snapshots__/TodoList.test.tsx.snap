// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="undefined"
  >
    <span
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-square fa-w-14 "
        data-icon="square"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    some todo
    <span
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-trash fa-w-14 "
        data-icon="trash"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm415.2 56.7L394.8 467c-1.6 25.3-22.6 45-47.9 45H101.1c-25.3 0-46.3-19.7-47.9-45L32.8 140.7c-.4-6.9 5.1-12.7 12-12.7h358.5c6.8 0 12.3 5.8 11.9 12.7z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;
